### ISSUE TITLE
preliminary result

### ISSUE THREAD

**Issue by rshu** [rshu/BugReportPrediction#1]
derby-train.csv derby-test.csv

C:\Users\terry\AppData\Local\Programs\Python\Python37\lib\site-packages\sklearn\externals\joblib\externals\cloudpickle\cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
              org    security     ...           check       label
count  500.000000  500.000000     ...      500.000000  500.000000
mean     3.600000    0.082000     ...        0.090000    0.092000
std     14.141569    0.442343     ...        0.560829    0.289315
min      0.000000    0.000000     ...        0.000000    0.000000
25%      0.000000    0.000000     ...        0.000000    0.000000
50%      0.000000    0.000000     ...        0.000000    0.000000
75%      1.000000    0.000000     ...        0.000000    0.000000
max    205.000000    5.000000     ...        8.000000    1.000000

[8 rows x 101 columns]
             org   security     ...           check       label
count  500.00000  500.00000     ...      500.000000  500.000000
mean     5.75200    0.03200     ...        0.048000    0.084000
std     14.03897    0.26668     ...        0.271742    0.277666
min      0.00000    0.00000     ...        0.000000    0.000000
25%      0.00000    0.00000     ...        0.000000    0.000000
50%      0.00000    0.00000     ...        0.000000    0.000000
75%      4.00000    0.00000     ...        0.000000    0.000000
max    124.00000    4.00000     ...        3.000000    1.000000

[8 rows x 101 columns]
Train_x Shape ::  (500, 100)
Train_y Shape ::  (500,)
Test_x Shape ::  (500, 100)
Test_y Shape ::  (500,)

----------naive bayes----------
Trained model: GaussianNB(priors=None, var_smoothing=1e-09)
Train Accuracy ::  0.95
Test Accuracy  ::  0.922
Confusion matrix [[442  16]
 [ 23  19]]
Precision  ::  0.7466973886328725
f1 score ::  0.7256264861898665
recall score :: 0.7087232272821793
              precision    recall  f1-score   support

           0       0.95      0.97      0.96       458
           1       0.54      0.45      0.49        42

   micro avg       0.92      0.92      0.92       500
   macro avg       0.75      0.71      0.73       500
weighted avg       0.92      0.92      0.92       500

----------random forest----------
Trained model: RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
            max_depth=None, max_features='auto', max_leaf_nodes=None,
            min_impurity_decrease=0.0, min_impurity_split=None,
            min_samples_leaf=1, min_samples_split=2,
            min_weight_fraction_leaf=0.0, n_estimators=10, n_jobs=None,
            oob_score=False, random_state=None, verbose=0,
            warm_start=False)
Train Accuracy ::  0.992
Test Accuracy  ::  0.92
Confusion matrix [[454   4]
 [ 36   6]]
Precision  ::  0.763265306122449
f1 score ::  0.5942875689711132
recall score :: 0.5670617592014972
              precision    recall  f1-score   support

           0       0.93      0.99      0.96       458
           1       0.60      0.14      0.23        42

   micro avg       0.92      0.92      0.92       500
   macro avg       0.76      0.57      0.59       500
weighted avg       0.90      0.92      0.90       500

----------KNN----------
Trained model: KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
           metric_params=None, n_jobs=None, n_neighbors=1, p=2,
           weights='uniform')
Train Accuracy ::  1.0
Test Accuracy  ::  0.912
Confusion matrix [[450   8]
 [ 36   6]]
Precision  ::  0.6772486772486772
f1 score ::  0.5838377723970943
recall score :: 0.5626949469744229
              precision    recall  f1-score   support

           0       0.93      0.98      0.95       458
           1       0.43      0.14      0.21        42

   micro avg       0.91      0.91      0.91       500
   macro avg       0.68      0.56      0.58       500
weighted avg       0.88      0.91      0.89       500

----------logistic regression----------
Trained model: LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='warn',
          n_jobs=None, penalty='l2', random_state=None, solver='warn',
          tol=0.0001, verbose=0, warm_start=False)
Train Accuracy ::  0.986
Test Accuracy  ::  0.922
Confusion matrix [[451   7]
 [ 32  10]]
Precision  ::  0.7609913530629644
f1 score ::  0.6487688899295737
recall score :: 0.6114056976502391
              precision    recall  f1-score   support

           0       0.93      0.98      0.96       458
           1       0.59      0.24      0.34        42

   micro avg       0.92      0.92      0.92       500
   macro avg       0.76      0.61      0.65       500
weighted avg       0.90      0.92      0.91       500

----------multilayer perceptron----------
Trained model: MLPClassifier(activation='relu', alpha=0.0001, batch_size='auto', beta_1=0.9,
       beta_2=0.999, early_stopping=False, epsilon=1e-08,
       hidden_layer_sizes=(100,), learning_rate='constant',
       learning_rate_init=0.001, max_iter=200, momentum=0.9,
       n_iter_no_change=10, nesterovs_momentum=True, power_t=0.5,
       random_state=None, shuffle=True, solver='adam', tol=0.0001,
       validation_fraction=0.1, verbose=False, warm_start=False)
Train Accuracy ::  0.998
Test Accuracy  ::  0.92
Confusion matrix [[454   4]
 [ 36   6]]
Precision  ::  0.763265306122449
f1 score ::  0.5942875689711132
recall score :: 0.5670617592014972
              precision    recall  f1-score   support

           0       0.93      0.99      0.96       458
           1       0.60      0.14      0.23        42

   micro avg       0.92      0.92      0.92       500
   macro avg       0.76      0.57      0.59       500
weighted avg       0.90      0.92      0.90       500

Process finished with exit code 0
